ament_tools/build_types/cmake.py

```diff
@@ -143,6 +143,7 @@ def on_build(self, context):
         # Calculate any extra cmake args which are not common between cmake build types
         extra_cmake_args = []
         if should_run_configure:
+            extra_cmake_args += ['-DBUILD_TESTING=%d' % int(context.build_tests)]
```

Comment:
Shouldn't this be outside the `should_run_configure` condition? Otherwise it wouldn't change when the `--build-tests` is toggled.

Reply:
Not sure, I was copying `ament_cmake`:

https://github.com/ament/ament_tools/blob/6294c63fdd2a55bc4a48ffcf937cb0b6a0fb19d9/ament_tools/build_types/ament_cmake.py#L98-L99

As I read it, the content of the `extra_cmake_args` are only important if `should_run_configure` is already true. Moving this out of that if conditional shouldn't change the behavior I think. As to why it should not be added to `extra_cmake_args` if configure won't be done, I don't know.

Reply:
The triggering of cmake is already based on `--build-tests` here:

https://github.com/ament/ament_tools/blob/6294c63fdd2a55bc4a48ffcf937cb0b6a0fb19d9/ament_tools/build_types/ament_cmake.py#L84

Maybe I don't understand what you're asking though.

Reply:
Ok, I see that the same check is already in place for plain CMake packages: https://github.com/ament/ament_tools/pull/184/files#diff-9e94d1bd2a8b57f7c156534b14d8ad3fR131

Sounds good to me than as-is.